### PR TITLE
Add highlighting for a Python console session or doctest with ">>>" and "..." at beginning of lines

### PR DESCRIPTION
**Pull request by msabramo**
Grammar for Python console (`pycon` lexer in Pygments) and doctest.

This is basically Python code that has `>>>` or `...` at the beginning of lines.

Refs: https://github.com/github/linguist/issues/1939

Cc: @aroben

### CODE REVIEW

**Comment by msabramo** [comment]
Here's the JSON version that I'm using in Atom 0.165.0

```json
{
  "scopeName": "text.python.console",
  "name": "Python Console",
  "fileTypes": [
    "pycon"
  ],
  "patterns": [
    {
      "match": "^(>{3}|\\.{3}) (.+)$",
      "captures": {
        "1": {
          "name": "punctuation.separator.prompt.python-session"
        },
        "2": {
          "patterns": [
            {
              "include": "source.python"
            }
          ]
        }
      }
    }
  ]
}
```

File: grammars/python-console.cson [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+'name': 'Python Console'
+'fileTypes': [
+  'pycon'
+]
+'patterns': [
+  {
+    'match': '^(>{3}|\\.{3}|In \\[\\d+\\]:) (.+)$'
+    'captures':
+      '1':
+        'name': 'punctuation.separator.prompt.python-session'
```
Comment (aroben) [inline grammars/python-console.cson]: I think this should be `punctuation.separator.prompt.python.console` to match the grammar's scope name.

**Comment by aroben** [comment]
This looks pretty good to me but could use some tests. You can look at https://github.com/atom/language-python/blob/master/spec/python-spec.coffee for some examples. (I'm not a maintainer of this repo, so take my suggestions with a grain of salt.)

**Comment by kevinsawicki** [comment]
Thanks for adding this :shipit:

**Comment by msabramo** [comment]
Awesome!

How long does it typically take for this to show up on GitHub?

**Comment by aroben** [comment]
@msabramo Changes usually show up within a couple of weeks.

**Comment by ionelmc** [comment]
Does this include any specific coloring for the traceback part (as in pygments) ?

**Comment by aroben** [comment]
I don't think so. The grammar would need to `'include': 'text.python.traceback'` somewhere.

**Pull request merged**